### DISCUSSION TITLE
Changing returned media type

### DISCUSSION THREAD

**Discussion by kabrannov** [specmatic/specmatic#1004] [General]
Hello, is it possible to force Specmatic to return endpoint response body in a format of my choice? I have an endpoint that can return avro/binary or avro/json. How can I set up specmatic to use the avro/json? Tried to set it up through OpenAPI specification but that didnt work.

**Comment by harikrishnan83** [comment]
@kabrannov Thanks for reaching out. Can you please share the OpenAPI spec you tried it with (or a similar file)? so that we can check? Meanwhile I will also try to setup a sample for this.

**Reply by kabrannov** [reply]
Hi @harikrishnan83 
The OpenAPI spec that I use is in the attachment, with confidential parts changed to dummy values. I tried to prepare the spec according to OpenAPI documentation but I cannot get specmatic to use my accept header. Every time I try this I am getting a random json as an Accept header, for example this one: 
 "Accept: {"UCTKR":"LBQXIOSXDG"}"
[OpenAPISpec.txt](https://github.com/znsio/specmatic/files/14447953/OpenAPISpec.txt)

**Reply by kabrannov** [reply]
@harikrishnan83 did you manage to get any sample openAPI spec with avro/json header?

**Reply by harikrishnan83** [reply]
@kabrannov not yet, will look into this shortly. Thanks for the reminder.
